src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -321,101 +321,6 @@ class AdjustClipBorder final : public WaveClipAdjustBorderHandle::AdjustPolicy
-{
-   std::pair<double, double> mRange;
-   std::vector<std::shared_ptr<WaveClip>> mLeftClips;
-   std::vector<std::shared_ptr<WaveClip>> mRightClips;
-   double mInitialBorderPosition{};
-   int mDragStartX{ };
-
-   void TrimTo(double t)
-   {
-      t = std::clamp(t, mRange.first, mRange.second);
-
-      for (auto& clip : mLeftClips)
-         clip->TrimRightTo(t);
-      for (auto& clip : mRightClips)
-         clip->TrimLeftTo(t);
-   }
-
-public:
-   AdjustBetweenBorders(
-      WaveTrack* track,
-      std::shared_ptr<WaveClip>& leftClip,
-      std::shared_ptr<WaveClip>& rightClip)
-   {
-      auto clips = track->GetClips();
-
-      wxASSERT(std::find(clips.begin(), clips.end(), leftClip) != clips.end());
-      wxASSERT(std::find(clips.begin(), clips.end(), rightClip) != clips.end());
-
-      if (track->IsAlignedWithLeader() || track->GetLinkType() == Track::LinkType::Aligned)
-      {
-         //find clips in other channels whose border should be also adjusted
-         mLeftClips = FindClipsInChannels(leftClip->GetPlayStartTime(), leftClip->GetPlayEndTime(), track);
-         mRightClips = FindClipsInChannels(rightClip->GetPlayStartTime(), rightClip->GetPlayEndTime(), track);
-      }
-      else
-      {
-         mLeftClips.push_back(leftClip);
-         mRightClips.push_back(rightClip);
-      }
-
-      mRange = std::make_pair(
-         //not less than 1 sample length
-         mLeftClips[0]->GetPlayStartTime() + 1.0 / mLeftClips[0]->GetRate(),
-         mRightClips[0]->GetPlayEndTime() - 1.0 / mRightClips[0]->GetRate()
-      );
-      mInitialBorderPosition = mRightClips[0]->GetPlayStartTime();
-   }
-
-   bool Init(const TrackPanelMouseEvent& event) override
-   {
-      if (event.event.LeftDown())
-      {
-         mDragStartX = event.event.GetX();
-         return true;
-      }
-      return false;
-   }
-
-   UIHandle::Result Drag(const TrackPanelMouseEvent& event, AudacityProject& project) override
-   {
-      const auto newX = event.event.GetX();
-      const auto dx = newX - mDragStartX;
-
-      auto& viewInfo = ViewInfo::Get(project);
-
-      auto eventT = viewInfo.PositionToTime(viewInfo.TimeToPosition(mInitialBorderPosition, event.rect.x) + dx, event.rect.x);
-      auto offset = sampleCount(
-         floor(
-            (eventT - mInitialBorderPosition) * mLeftClips[0]->GetRate()
-         )
-      ).as_double() / mLeftClips[0]->GetRate();
-
-      TrimTo(mInitialBorderPosition + offset);
-
-      return RefreshCode::RefreshCell;
-   }
-
-   void Finish(AudacityProject& project) override
-   {
-      if (mRightClips[0]->GetPlayStartTime() != mInitialBorderPosition)
-      {
-         auto dt = std::abs(mRightClips[0]->GetPlayStartTime() - mInitialBorderPosition);
-         ProjectHistory::Get(project).PushState(XO("Clip-Trim-Between"),
-               XO("Moved by %.02f").Format(dt));
-      }
-   }
-
-   void Cancel() override
-   {
-      TrimTo(mInitialBorderPosition);
-   }
-};
-
-
```

Comment:
But was there an intention to use this?  Was the design not completely realized?

Was the intention to drag the line between touching clips, and shrink one and grow the other together?

Reply:
Your guess is correct. Eventually, design team decided that we don't need that feature.